exemplar/src/main/kotlin/com/squareup/exemplar/actions/ChatWebSocketAction.kt

```diff
@@ -0,0 +1,52 @@
+package com.squareup.exemplar.actions
+
+import misk.web.ConnectWebSocket
+import misk.web.PathParam
+import misk.web.actions.WebAction
+import misk.web.actions.WebSocket
+import misk.web.actions.WebSocketListener
+import javax.inject.Singleton
+
+@Singleton
+class ChatWebSocketAction : WebAction {
+  private val rooms = mutableMapOf<String, Room>()
```

Comment:
this needs _something_ for concurrency. I think a Guava cache is the easiest thing to do self-creating map thingies.

``` 
  LoadingCache<Key, Graph> graphs = CacheBuilder.newBuilder()
       .build(
           new CacheLoader<Key, Graph>() {
             public Graph load(Key key) throws AnyException {
               return createExpensiveGraph(key);
             }
           });
```